### ISSUE TITLE
auto craft update

### ISSUE THREAD

**Issue by wibi9424** [Zergatul/cheatutils#106]
ive turn on fireworks rocket, but the flight duration is 1, which i wanted is 3, can you add more advance in autocraft things,

can you turn on chat in this github, maybe other github member will answer my question about scripting, see in my github repository everybody can chat without create new issue.

**Comment by Zergatul** [comment]
I don't think anyone is really sitting here in github and check issues/chat (besides me).
You can join matrix and ask about scripts.

I will check fireworks, but from what I remember this recipe for lvl 3 fireworks is not displayed inside crafting menu. Maybe I will need to hardcode this exact scenario.

**Comment by Zergatul** [comment]
I enabled "discussions"

**Comment by Zergatul** [comment]
As I thought there is a problem with fireworks. Proper firework recipe exists only on the server side. But it is CustomRecipe meaning it has custom logic to generate crafting output because of firework stars and gunpowder amount.
So, really the simplest option could be to just hardcode fireworks are craft level 3.

**Comment by wibi9424** [comment]
yeah fireworks has many variations, that maybe in future can be toggled, like firework shape, or color, or else that maybe hard to do,

sometimes we need just flight duration to 3 using 1 paper and 3 gun powder just for elytra things, sometimes else we need so many kind of fireworks that for celebration or else,

but I think in this new Minecraft version no need to hurry coding on this, because Minecraft also have vanilla auto crafter, i just need to see how it works, maybe it hard in begining to create redstone building or maybe Minecraft server has redstone limit which i must find another way, but after that it will easy to use.

and thanks for adding discussion, now i will not create new issue for my suggestions for this mod.

**Comment by Zergatul** [comment]
Yeah, I abandoned the idea of supporting level 3 fireworks. The problem is current code calls place recipe method, and the game moves ingredients automatically:
```java
RecipeHolder<CraftingRecipe> holder = findRecipe(config);
if (holder != null) {
    boolean shift = holder.value().getResultItem(mc.level.registryAccess()).getItem().getDefaultMaxStackSize() > 1;
    mc.gameMode.handlePlaceRecipe(craftingScreen.getMenu().containerId, holder, shift);
    state = State.RECIPE_CLICKED;
}
```
And since there is no recipe for custom fireworks I would need to code completely different approach.

**Comment by wibi9424** [comment]
i hope i can code java, the syntax kinda manny things called for simple task so far i just know PHP language, because i can create some little application that run in browser you can visit my github to see it, java and c++ is my next language, for me coding is just for fun or hobby, thanks because to create this mod, this mod is the one mod that i must have, second is sodium or optifine.
